### PR TITLE
HLM-5359:: Referral Reconciliation Extraction

### PR DESCRIPTION
**Pull request by Ramkrishna-egov**

### CODE REVIEW

File: apps/health_campaign_field_worker_app/lib/pages/beneficiary_registration/individual_details.dart [hunk 1]
```diff
@@ -644,6 +644,7 @@ class _IndividualDetailsPageState
                                     ),
                                     onPressed: () {
                                       Navigator.of(context).push(
+                                        // [TODO: Add the route to auto_route]
```
Comment (naveen-egov) [inline apps/health_campaign_field_worker_app/lib/pages/beneficiary_registration/individual_details.dart:647]: Please update the TODO

File: packages/referral_reconciliation/LICENSE [hunk 2]
```diff
@@ -0,0 +1 @@
+TODO: Add your license here.
```
Comment (naveen-egov) [inline packages/referral_reconciliation/LICENSE:1]: Add the License Details

File: packages/referral_reconciliation/lib/blocs/search_referral_reconciliations.dart [hunk 3]
```diff
@@ -82,18 +71,19 @@ class SearchReferralsBloc
       searchQuery: event.searchText,
     ));
 
-    List<HFReferralModel> hfReferrals = await hfReferralDataRepository.search(
-      HFReferralSearchModel(
-        projectId: event.projectId,
-        name: event.searchText,
-      ),
-    );
+    // Fetch referrals based on the name.
```
Comment (naveen-egov) [inline packages/referral_reconciliation/lib/blocs/search_referral_reconciliations.dart:74]: Remove un-used code

File: packages/referral_reconciliation/lib/pages/create_referral/create_hf_referral_wrapper.dart [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    this.viewOnly = false,
+    this.referralReconciliation,
+    required this.cycles,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    if (cycles.isEmpty) {
+      return const Center(
+        child: Text('No Cycles exists for the project'),
```
Comment (naveen-egov) [inline packages/referral_reconciliation/lib/pages/create_referral/create_hf_referral_wrapper.dart:30]: Move to constants and use i18 code

File: packages/referral_reconciliation/lib/pages/create_referral/reason_checklist_preview.dart [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+      body: ScrollableContent(
+        header: const Column(children: [
+          BackNavigationHelpHeaderWidget(),
+        ]),
+        footer:
+            BlocBuilder<ReferralReconServiceBloc, ReferralReconServiceState>(
+          builder: (context, state) {
+            return state.maybeWhen(
+              orElse: () => const Offstage(),
+              serviceSearch: (value1, value2, value3) {
```
Comment (naveen-egov) [inline packages/referral_reconciliation/lib/pages/create_referral/reason_checklist_preview.dart:42]: Provide value specifiers

File: packages/referral_reconciliation/lib/pages/search_referral_reconciliations.dart [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+                                      } else {
+                                        bloc.add(SearchReferralsByNameEvent(
+                                          searchText: value.trim(),
+                                        ));
+                                      }
+                                    },
+                                  ),
+                                ],
+                              ),
+                              const SizedBox(height: 16),
```
Comment (naveen-egov) [inline packages/referral_reconciliation/lib/pages/search_referral_reconciliations.dart:151]: use Kpadding

File: packages/referral_reconciliation/lib/widgets/back_navigation_help_header.dart [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+                          i18.common.coreCommonBack,
+                        ),
+                        overflow: TextOverflow.ellipsis,
+                      ),
+                    ),
+                  ),
+              ],
+            ),
+          ),
+          SizedBox(width: showHelp ? 16 : 0),
```
Comment (naveen-egov) [inline packages/referral_reconciliation/lib/widgets/back_navigation_help_header.dart:57]: use kpadding

File: packages/referral_reconciliation/lib/widgets/beneficiary_card.dart [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+              : DigitIconButton(
+                  icon: Icons.info_rounded,
+                  iconText: ReferralReconLocalization.of(context)
+                      .translate(status.toString()),
+                  iconTextColor: theme.colorScheme.error,
+                  iconColor: theme.colorScheme.error,
+                ),
+        ),
+        Padding(
+          padding: const EdgeInsets.all(4),
```
Comment (naveen-egov) [inline packages/referral_reconciliation/lib/widgets/beneficiary_card.dart:57]: use kpadding

File: packages/referral_reconciliation/test/referral_reconciliation_test.dart [hunk 9]
```diff
@@ -0,0 +1 @@
+void main() {}
```
Comment (naveen-egov) [inline packages/referral_reconciliation/test/referral_reconciliation_test.dart:1]: Please add the necessary Test cases

**Review by naveen-egov — CHANGES_REQUESTED** [review]
Please make the necessary requested changes

File: packages/referral_reconciliation/test/referral_reconciliation_test.dart (hunk 9 above)
Comment (Ramkrishna-egov) [reply packages/referral_reconciliation/test/referral_reconciliation_test.dart:1]: Test cases will be added in [HLM-5567](https://digit-discuss.atlassian.net/browse/HLM-5567)

File: packages/referral_reconciliation/lib/blocs/search_referral_reconciliations.dart (hunk 3 above)
Comment (Ramkrishna-egov) [reply packages/referral_reconciliation/lib/blocs/search_referral_reconciliations.dart:74]: It is code comments

**Review by naveen-egov — APPROVED** [review]

**Pull request merged**